### PR TITLE
Answer and Result Module

### PR DESCRIPTION
**Pull request by mmkarim**

### CODE REVIEW

File: app/views/categories/_category.html.erb [hunk 1]
```diff
@@ -1,4 +1,16 @@
...
-  <%= link_to category.title, category  %>
-  <p> <%= category.info %> </p>      
+  <div class="form-inline">
+    <div class="form-group">
+      <h3> <%= category.title %> </h3>
+      <p> <%= category.info %> </p>
+    </div>
+    <% if logged_in? %>
+      <div class="form-group">
+          <%= form_for Lesson.new do |f| %>
```
Comment (dieunb) [inline app/views/categories/_category.html.erb]: indent

File: app/views/lessons/_result_modal.html.erb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      <div class="modal-header">
+        <button type="button" class="close" data-dismiss="modal"
+          aria-label="Close">
+          <span aria-hidden="true">&times;</span>
+        </button>
+        <h4 class="modal-title" id="myModalLabel">
+          <%= t :result %>
+        </h4>
+      </div>
+      <% correct = 0 %>
```
Comment (dieunb) [inline app/views/lessons/_result_modal.html.erb]: should write a function to get correct number

File: app/controllers/lessons_controller.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    @lesson = Lesson.find_by id: params[:id]
+    @lesson.update_attributes lesson_params
+
+    respond_to do |format|
+      format.js
+    end
+  end
+
+  private
+  def authorize_user
```
Comment (chuanhtuan) [inline app/controllers/lessons_controller.rb]: write in SessionHelper

File: app/models/lesson.rb [hunk 4]
```diff
@@ -1,4 +1,21 @@
 class Lesson < ActiveRecord::Base
   belongs_to :user
   belongs_to :category
+
+  has_many :words, through: :results
+  has_many :results
```
Comment (chuanhtuan) [inline app/models/lesson.rb]: move to up 1 line :(

File: app/models/lesson.rb [hunk 5]
```diff
@@ -1,4 +1,21 @@
...
+
+  before_create :assign_words
+
+  def number_correct_choices
+    results.select{|result| result.choice.try :correct}.count
+  end
+
+  private
+  def assign_words
+    words = self.category.words.order("RANDOM()").limit(10)
```
Comment (chuanhtuan) [inline app/models/lesson.rb]: 10 should be defined in Setting

File: app/views/lessons/_result_modal.html.erb [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        <table class="table table-bordered table-striped">
+          <thead>
+            <tr>
+              <th><%= t"word" %></th>
+              <th><%= t"option" %></th>
+              <th><%= t"result" %></th>
+            </tr>
+          </thead>
+          <tbody>
+            <% @lesson.results.each do |r| %>
```
Comment (chuanhtuan) [inline app/views/lessons/_result_modal.html.erb]: <% @lesson.results.each do |result| %>

File: config/routes.rb [hunk 7]
```diff
@@ -11,7 +11,10 @@
 
   resources :relationships, only: [:create, :destroy]
 
-  get "lessons" => "categories#index"
+  resources :users
+
```
Comment (chuanhtuan) [inline config/routes.rb]: remove empty line

File: config/routes.rb [hunk 8]
```diff
@@ -11,7 +11,10 @@
...
+  resources :lessons
+
```
Comment (chuanhtuan) [inline config/routes.rb]: remove empty line

File: config/routes.rb [hunk 9]
```diff
@@ -11,7 +11,10 @@
...
   resources :categories, only: [:show, :index]
```
Comment (chuanhtuan) [inline config/routes.rb]: remove empty line

File: app/views/lessons/show.html.erb [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+          <%= result_builder.radio_button :choice_id, choice.id %>
+          <%= choice.content %> <%= "true" if choice.correct? %>
+        </p>
+      <%end%>
+    </div>
+  <% end %>
+  <%= f.submit t("submit"), class: "btn btn-primary" %>
+<% end %>
+
+<%= link_to "<< Previous", "javascript:void(0)", id: "previous", class: "pull-left" %>
```
Comment (dieunb) [inline app/views/lessons/show.html.erb]: i18n

File: app/views/lessons/show.html.erb [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+<%= link_to "Next >>", "javascript:void(0)", id: "next", class: "pull-right" %>
```
Comment (dieunb) [inline app/views/lessons/show.html.erb]: i18n

File: app/models/lesson.rb [hunk 12]
```diff
@@ -1,4 +1,20 @@
 class Lesson < ActiveRecord::Base
   belongs_to :user
   belongs_to :category
+  has_many :words, through: :results
+  has_many :results
```
Comment (chuanhtuan) [inline app/models/lesson.rb]: has_many :results
has_many :words, through: :results

File: config/routes.rb [hunk 13]
```diff
@@ -10,10 +10,9 @@
   end
 
   resources :relationships, only: [:create, :destroy]
-
-  get "lessons" => "categories#index"
+  resources :users
+  resources :lessons
```
Comment (chuanhtuan) [inline config/routes.rb]: should use nested resource
resources :categories, only: [:show, :index] do
  resources :lessons
end

File: app/views/categories/_category.html.erb [hunk 14]
```diff
@@ -1,4 +1,16 @@
...
-  <%= link_to category.title, category  %>
-  <p> <%= category.info %> </p>      
+  <div class="form-inline">
+    <div class="form-group">
+      <h3> <%= category.title %> </h3>
+      <p> <%= category.info %> </p>
+    </div>
+    <% if logged_in? %>
+      <div class="form-group">
+        <%= form_for Lesson.new do |f| %>
```
Comment (chuanhtuan) [inline app/views/categories/_category.html.erb]: <% form_for category, Lesson.new do |f| %>

File: app/models/word.rb [hunk 15]
```diff
@@ -1,6 +1,16 @@
 class Word < ActiveRecord::Base
   belongs_to :category
   has_many :choices, dependent: :destroy
+  has_many :lessons, through: :results
+  has_many :results
```
Comment (chuanhtuan) [inline app/models/word.rb]: has_many :results
has_many :lessons, through: :results

File: app/views/layouts/_header.html.erb [hunk 16]
```diff
@@ -5,7 +5,7 @@
       <ul class="nav navbar-nav navbar-right">
         <li><%= link_to t("home"), root_path %></li>
         <li><%= link_to t("help"), help_path %></li>
-        <li><%= link_to t("lessons"), lessons_path %></li>
+        <li><%= link_to t("lessons"), categories_path %></li>
```
Comment (chuanhtuan) [inline app/views/layouts/_header.html.erb]: category_lessons_path

Comment (mmkarim) [inline app/views/layouts/_header.html.erb]: I actually wanted to show the category... so may i change it with: 
<%= link_to t("category"), categories_path %>

**Pull request merged**